src/shared/ui/survey/CheckBox/CheckBoxActivityItem.tsx

```diff
@@ -0,0 +1,76 @@
+import { FC, useEffect, useState } from 'react';
+
+import { shuffle } from '@shared/lib';
+import { Box, ScrollView } from '@shared/ui';
+
+import CheckBoxItem from './CheckBoxItem';
+import { Item } from './types';
+
+type CheckBoxProps = {
+  // @todo make sure backend will update config to new keys described below (type RefactoredConfig in ./types)
+  config: {
+    itemList: Item[];
+    minValue: number;
+    maxValue: number;
+    colorPalette: boolean;
+    randomizeOptions: boolean;
+  };
+  onChange: (arrayOfValues: number[]) => void;
+  initialValues?: number[];
+};
+
+const CheckBoxActivityItem: FC<CheckBoxProps> = ({
+  config,
+  onChange,
+  initialValues = [],
+}) => {
+  const [items, setItems] = useState<Item[]>([]);
+  const [answer, setAnswer] = useState<number[]>(initialValues);
+  const { minValue, maxValue } = config;
+  const hasSingleItem = maxValue === 1 && minValue === 1;
+
+  useEffect(() => {
+    const { itemList, randomizeOptions } = config;
+    const filteredItems = itemList.filter(({ isVis }) => !isVis);
+    if (randomizeOptions) {
+      setItems(shuffle(filteredItems));
+    } else {
+      setItems(filteredItems);
+    }
+  }, [config]);
+
+  useEffect(() => {
+    onChange(answer);
+  }, [answer, onChange]);
```

Comment:
No need for this effect. Because `onChange` should be called right away inside `onItemValueChanged` call